runtime/onert/core/src/ir/Shape.cc

```diff
@@ -73,10 +73,12 @@ void Shape::extendRank(int to_rank)
 
 uint64_t Shape::num_elements() const
 {
+  constexpr int32_t unspecified_dim = -1;
+
   // if dimension is 0, it means unspecified and cannot calculate the total number of elements
   if (std::any_of(_dimensions.begin(), _dimensions.end(),
-                  [](const int32_t &v) { return (v == 0); }))
-    throw std::runtime_error("num_elements() cannot calculate when a dimension is unspecified");
+                  [](const int32_t &v) { return (v == unspecified_dim); }))
```

Comment:
@hyunsik-yoon (optional) (beyond PR) redundant parenthesis
```suggestion
                  [](const int32_t &v) { return v == unspecified_dim; }))
```